### PR TITLE
chore: debug sentinel docs issues

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Removes the unused `rivet-graphql` dependency, which was causing complications in ensuring `next` versions matched in the associated content deploy previews.

Also cleans up files related to the removal of `rivet-graphql`, such as `lib/cms`. Intent is to get things working, cleanup was not meant to be exhaustive.

## 🧪 Testing

- [ ] Visit the [preview]
    - In both Dev Dot and Sentinel mode (the latter accessed from the products dropdown in the lower left)...
    - The site should continue to function as expected
    - The layout should render without any errors
    - The [404] page should render as expected
        - **Please note**: if you view the [404] page in one of the preview modes first (Dev Dot or Sentinel), it will always appear in that preview mode in subsequent requests, even after changing the preview mode again. Visit a distinct 404 URL and you should see the correct layout for each 404 view.
- [ ] Verify that the PR https://github.com/hashicorp/sentinel/pull/722 has a working deploy preview
    - That PR uses this PR's branch to build a content deploy preview from the associated content repository

## 💭 Anything else?

Not at the moment.

[task]: https://app.asana.com/0/1202097197789424/1204836328648212/f
[preview]: https://dev-portal-git-zsdebug-sentinel-issues-hashicorp.vercel.app
[404]: https://dev-portal-git-zsdebug-sentinel-issues-hashicorp.vercel.app/this-is-a-404-page

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -190,7 +190,6 @@
 		"react-player": "^2.9.0",
 		"rehype-sanitize": "^5.0.1",
 		"remark-gfm": "^3.0.1",
-		"rivet-graphql": "^0.3.1",
```
Comment (zchsh) [inline package.json:193]: I've removed `rivet-graphql` - it was used for the `io` sites, but is no longer needed. This may seem unrelated to the Sentinel content deploy preview issue we're trying to target, but this removal seemed necessary to make the `next` version alignment work as expected.

File: src/lib/cms.ts [hunk 2]
```diff
@@ -13,21 +12,20 @@ import vagrant from 'data/vagrant.json'
...
+ *
+ * The typed `productConfig` in this file is still used in one place,
+ * in our `_error` page.
+ *
+ * TODO: clean up use of `{ productConfig }` named export from this file,
+ * maybe renaming this file (`product-config-dictionary`?) or doing something
+ * like an `fs` read of the `.json` data (maybe even validating with `zod`)
+ * rather than exporting a heavy-ish dictionary with *all* product data?
+ */
 const productConfig = [
```
Comment (zchsh) [inline src/lib/cms.ts:29]: I've left the `productConfig` export here as it's used on the `_error` page - I didn't want to dive too deep into a refactor. I've tried to add a clear comment so we can come back and clean this up later.

File: src/pages/_app.tsx [hunk 3]
```diff
@@ -104,14 +100,7 @@ export default function App({
...
 	 */
-	const allLayoutProps = {
-		theme: Component.theme,
-		...pageProps.layoutProps,
-		// @ts-expect-error - layoutProps is inferred from `fetchLayoutProps`,
-		// which current resolves to `unknown | null`.
-		// Spread types may only be created from object types.
-		...layoutProps,
-	}
+	const allLayoutProps = { theme: Component.theme, ...pageProps.layoutProps }
```
Comment (zchsh) [inline src/pages/_app.tsx:103]: I've removed `layoutProps` entirely. The variable name is a bit misleading - it really refers to specific `io` layout props, and only those props coming from Dato. The only "io"-ish layout left is Sentinel's, and it does _not_ require any data from Dato.

So, `layoutProps` seems very safe to remove (and as mentioned previously, somewhat necessary to make the `next`-version-parity fix easier by removing `rivet-graphql`.

File: src/pages/_error.jsx [hunk 4]
```diff
@@ -61,7 +56,7 @@ function Error({ statusCode, proxiedProductSlug, layoutProps }) {
 	const Layout = isProxiedDotIo ? ProxiedLayout : BaseLayout
 
 	return (
-		<Layout data={{ ...layoutProps }}>
```
Comment (zchsh) [inline src/pages/_error.jsx:64]: As with `_app.tsx`, the `io`-specific Dato-specific `layoutProps` are no longer needed here, as the only `io`-ish site we render is Sentinel, and it does not require data fetched from Dato.

File: src/pages/_app.tsx [hunk 5]
```diff
@@ -158,24 +147,7 @@ export default function App({
...
-	} else if (ctx.pathname.includes('_proxied-dot-io/nomad')) {
-		proxiedProduct = 'nomad'
-	} else if (ctx.pathname.includes('_proxied-dot-io/boundary')) {
-		proxiedProduct = 'boundary'
-	} else if (ctx.pathname.includes('_proxied-dot-io/packer')) {
-		proxiedProduct = 'packer'
-	} else if (ctx.pathname.includes('_proxied-dot-io/vagrant')) {
-		proxiedProduct = 'vagrant'
-	}
-	const layoutProps = await fetchLayoutProps(Component.layout, proxiedProduct)
```
Comment (BRKalow) [inline src/pages/_app.tsx:177]: wow nice cleanup here 🤩

File: src/lib/cms.ts [hunk 6]
```diff
@@ -13,21 +12,20 @@ import vagrant from 'data/vagrant.json'
...
+ * Dot-io sites have since been migrated to the `hashicorp/web` monorepo,
+ * so the Dato CMS related code was removed from this file.
+ *
+ * The typed `productConfig` in this file is still used in one place,
+ * in our `_error` page.
+ *
+ * TODO: clean up use of `{ productConfig }` named export from this file,
+ * maybe renaming this file (`product-config-dictionary`?) or doing something
+ * like an `fs` read of the `.json` data (maybe even validating with `zod`)
+ * rather than exporting a heavy-ish dictionary with *all* product data?
```
Comment (BRKalow) [inline src/lib/cms.ts:27]: 👍

**Review by BRKalow — APPROVED** [review]
Looks good to me! Followed your testing steps and it's working as expected. Great job doing some cleanup here as well 🙏
